carstore/read_only_blockstore.go

```diff
@@ -48,7 +48,7 @@ func (r *CarReadOnlyStoreTracker) GetOrCreate(key string, carFilePath string) (C
 		return bs, nil
 	}
 
-	rdOnly, err := blockstore.OpenReadOnly(carFilePath, carv2.ZeroLengthSectionAsEOF)
+	rdOnly, err := blockstore.OpenReadOnly(carFilePath, carv2.ZeroLengthSectionAsEOF(true))
```

Comment:
Do we need to set option `blockstore.UseWholeCIDs(true)` here?

Ditto in other places that instantiate CARv2 blockstore.

Reply:
Have done this here and at all the other places. Thanks !